kong/runloop/handler.lua

```diff
@@ -687,12 +688,26 @@ return {
   },
   header_filter = {
     before = function(ctx)
+      local var = ngx.var
+      local header = ngx.header
+
       if ctx.KONG_PROXIED then
         local now = get_now()
         -- time spent waiting for a response from upstream
         ctx.KONG_WAITING_TIME             = now - ctx.KONG_ACCESS_ENDED_AT
         ctx.KONG_HEADER_FILTER_STARTED_AT = now
       end
+
+      local upstream_status_header = constants.HEADERS.UPSTREAM_STATUS
+      if singletons.configuration.headers[upstream_status_header] then
+        local matches, err = re_match(var.upstream_status,
```

Comment:
It just struck me that we are missing a case here that would result in a Lua error (HTTP 500) when the request was short-circuited by a plugin (e.g. authentication). We should either nest this below the above `if proxied` branch, or program defensively against such a case. I will be adding a regression test and making that change in the merge.

Reply:
A tweaked version of this came to my mind if the code would insert this header for not proxied requests as well but seems like my understanding was wrong. Proxied request means that a request actually hits the upstream. Thank you for catching this!